src/aws-sdk/index.ts

```diff
@@ -0,0 +1,241 @@
+import { aws_iam } from "aws-cdk-lib";
+// eslint-disable-next-line import/no-extraneous-dependencies
+import * as AWS from "aws-sdk";
+import { ASLGraph } from "../asl";
+import { ErrorCodes, SynthError } from "../error-code";
+import { Expr, Argument } from "../expression";
+import {
+  isArgument,
+  isObjectLiteralExpr,
+  isStringLiteralExpr,
+  isArrayLiteralExpr,
+  isPropAssignExpr,
+} from "../guards";
+import { makeIntegration } from "../integration";
+import { evalToConstant } from "../util";
+import { SDK_INTEGRATION_SERVICE_NAME } from "./asl";
+import { IAM_SERVICE_PREFIX } from "./iam";
+import type { SDK as TSDK, ServiceKeys, SdkCallInput } from "./types";
+
+export const SDK: TSDK = new Proxy<any>(AWS, {
+  isExtensible() {
+    return false;
+  },
+  setPrototypeOf() {
+    return false;
+  },
+  set() {
+    return false;
+  },
+  get(_, serviceName: ServiceKeys) {
+    return new ServiceProxy(serviceName);
+  },
+});
+
+class ServiceProxy {
+  constructor(serviceName: ServiceKeys) {
+    return new Proxy(new AWS[serviceName](), {
+      isExtensible() {
+        return false;
+      },
+      setPrototypeOf() {
+        return false;
+      },
+      set() {
+        return false;
+      },
+      get: (_, methodName: string) => {
+        return makeSdkIntegration(serviceName, methodName);
+      },
+    });
+  }
+}
+
+function makeSdkIntegration(serviceName: ServiceKeys, methodName: string) {
+  return makeIntegration<
+    `$AWS.SDK.${ServiceKeys}`,
+    (input: any) => Promise<any>
+  >({
```

Comment:
Note: you can also implement the `Integration` interface like `StepFunction`, `Function`, etc does if you'd prefer a class. Up to you.

The `makeIntegration` method is cleaner, but limited.